src/node/history.h

```diff
@@ -251,6 +251,7 @@ namespace ccf
     kv::Consensus::SignableTxIndices commit_txid;
     kv::Store& store;
     T& replicated_state_tree;
+    SpinLock& state_lock; // lock to protect replicated_state_tree
```

Comment:
From an encapsulation perspective, I would prefer if we passed a ref to the TxHistory, and called get_replicated_state_root() on it (which is already synchronised), rather than the raw tree and the lock along with it.